test/geometry/LinearPropagator.test.cu

```diff
@@ -47,7 +47,7 @@ __global__ void linProp_test_kernel(const GeoParamsPointers shared,
         distances[tid.get() * max_segments + seg] = geo.next_step();
 
         // Move next step
-        propagate();
+        auto temp = propagate();
```

Comment:
This whole  test needs to be updated I think. Right now it's not really testing the propagator on GPU. The inside of the loop should be something like:
```
auto step = propagate();
ids[tid.get() * max_segments + seg]       = step.volume.unchecked_get();
distances[tid.get() * max_segments + seg] = step.distance();
if (!step.volume)
    break;
```

Reply:
Good catch, suggestions taken.